### DISCUSSION TITLE
Lesson 4: I installed web3 successfully but I couldn't deploy.

### DISCUSSION THREAD

**Discussion by HalaAura** [smartcontractkit/full-blockchain-solidity-course-py#547] [Q&A] [answered]
Hello! I installed web3 successfully and when I wrote:
 from web3 import Web3
web3 was underlined with yellow line and it gave me an error when I run python deploy.py AFTER I changed file name from web3_py_simple_storage to web3.py it worked but then I still get an error in the message below, what should I do? 

Here is my code: 

```Python
from solcx import compile_standard, install_solc
import solcx
import json
from web3 import Web3

solcx.install_solc("0.8.7")  # or whatever version we want
with open("SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
    print("Installing...")
    install_solc("0.6.0")

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"simpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)

# print(compiled_sol)
with open("compiled_code.json", "w") as file:
    json.dump(compiled_sol, file)

# get bytecode
bytecode = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["evm"][
    "bytecode"
]["object"]

# get abi
abi = json.loads(
    compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["metadata"]
)["output"]["abi"]


w3 = Web3(Web3.HTTPProvider("http://127.0.0.1:7545"))
chain_id = 5777
my_address = "0x3B7468eF9172d14001878C4181D846d3A1320f35"
private_key = "0x0a3667b39f3fed7a299c615063932f63c96395815e19177165820163ecf0994a"

SimpleStorage = w3.eth.contract(abi=abi, bytecode=bytecode)
print(SimpleStorage)

```

The error message: 
INFO: Could not find files for the given pattern(s).
Installing...
Traceback (most recent call last):
  File "C:\Users\user\web3.py\deploy.py", line 30, in <module>
    bytecode = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["evm"][
KeyError: 'SimpleStorage.sol'

**Comment by cromewar** [comment, marked as answer]
Hello @HalaAura, in the line ` "sources": {"simpleStorage.sol": {"content": simple_storage_file}},` the line is SimpleStorage.sol with upper case.

Best regards, Cromewar.

**Reply by HalaAura** [reply]
Should I do something for this part of the message: INFO: Could not find files for the given pattern(s)?

**Comment by HalaAura** [comment]
Thank you so much!! :) I edited it and now this is the output: 
INFO: Could not find files for the given pattern(s).
Installing...
<class 'web3._utils.datatypes.Contract'>

**Reply by cromewar** [reply]
This is Ok
`INFO: Could not find files for the given pattern(s).` this is not an actual error, so you don't have anything to worry about.
And
`<class 'web3._utils.datatypes.Contract'>` is also ok, as you are printing `print(SimpleStorage)` if you want to get the address you should print `print(SimpleStorage.address)`.

**Reply by HalaAura** [reply]
Great!! Thanks a lot :)
